### ISSUE TITLE
Ignore which slot a tool is in.

### ISSUE THREAD

**Issue by ghost** [codetaylor/artisan-worktables#210]
Is it possible to get the recipes to ignore which slot a tool is in? For example, when manually crafting, sometimes the hammer is in the top slot and the trowel is in the middle. If the hammer is called for, it ignores the trowel, but if the trowel is called for the recipe doesn't work.

**Comment by codetaylor** [comment]
I thought that was fixed in `1.22.7`.

![image](https://user-images.githubusercontent.com/1240716/68079861-d10d7380-fdad-11e9-87a9-2d4b7a59a6d3.png)

What version are you on?

Can you provide an example of what you're talking about?

Do you have example recipes?

**Comment by ghost** [comment]
1.12.2-1.22.9 (Seablock Modpack 1.1.6).
I admit to not checking with a newer version because the recipes are all scripted.
Also, some recipes appear to work regardless of the tool position. I'll mess with it some more tomorrow if you like and see if I can find a pattern.

<img width="665" alt="Screen Shot 2019-11-02 at 9 48 53 PM" src="https://user-images.githubusercontent.com/15851438/68080531-6718bc80-fdf5-11e9-8155-5c1463cf6651.png">
<img width="664" alt="Screen Shot 2019-11-02 at 9 48 13 PM" src="https://user-images.githubusercontent.com/15851438/68080532-67b15300-fdf5-11e9-9e57-2434501f1a9b.png">

**Comment by codetaylor** [comment]
What are the tool requirements for the recipe images you posted? Which tool does it require? Both?

**Comment by codetaylor** [comment]
I have been able to reproduce this using the following recipe:

```java
RecipeBuilder.get("basic")
  .setShapeless([<minecraft:gravel>])
  .addOutput(<minecraft:flint>)
  .addTool(<ore:artisansCutters>, 48)
.create();
```

![image](https://user-images.githubusercontent.com/1240716/68089165-e53b8a00-fe1a-11e9-8e77-ee5088e54469.png)

![image](https://user-images.githubusercontent.com/1240716/68089175-0c925700-fe1b-11e9-9fd6-16bb696a5f0d.png)

**Comment by codetaylor** [comment]
This *should* be solved in `1.22.13`.

In retrospect, the tool matching should never have been treated as a bug and "fixed" in a patch version. It was never a bug to begin with because it was designed with the intention that tools would be restricted to the slots that the recipe defined and that design was foundational to several of the algorithms regarding tool matching.

It is clear now the change was not trivial and the ramifications of that change wasn't fully understood when it was made. This is a change that should have been made in a feature version.
